loki/client_test.go

```diff
@@ -323,3 +324,33 @@ func createServerHandler(receivedReqsChan chan receivedReq, status int) http.Han
 		rw.WriteHeader(status)
 	})
 }
+
+type roundTripFunc func(r *http.Request) (*http.Response, error)
+
+func (s roundTripFunc) RoundTrip(r *http.Request) (*http.Response, error) {
+	return s(r)
+}
+
+func TestClient_EncodeJSON(t *testing.T) {
+	c, err := NewWithDefault("http://loki.com/loki/api/v1/push")
+	require.NoError(t, err)
+	c.cfg.EncodeJson = true
+
+	c.client.Transport = roundTripFunc(func(r *http.Request) (*http.Response, error) {
+		require.Equal(t, r.Header["Content-Type"][0], JSONContentType)
```

Comment:
we can replace it with `r.Header.Get("Content-Type")`!

Reply:
Yes we can !